src/components/cylc/GScan.vue

```diff
@@ -0,0 +1,204 @@
+          name
+          state
+          cyclePoint
+          latestMessage
+          task {
+            meanElapsedTime
+            name
+          }
+          jobs(sort: { keys: ["submit_num"], reverse:true }) {
+            id
+            batchSysName
+            batchSysJobId
+            host
+            startedTime
+            submittedTime
+            finishedTime
+            state
+            submitNum
+          }
+        }
+      }
+    }
+  `
+}
+
+export default {
+  name: 'GScan',
+  components: {
+    job: Job
+  },
+  props: {
+    /**
+     * Vanilla workflows object from GraphQL query
+     */
+    workflows: {
+      type: Array,
+      required: true
+    }
+  },
+  data () {
+    return {
+      viewID: '',
+      subscriptions: {},
+      isLoading: true
+    }
+  },
+  computed: {
+    /**
+     * Compute summary information, where the key is the name of a workflow, the value is another map with the summary.
+     * @returns {Map<String, Map>}
+     */
+    workflowsSummaries () {
+      const workflowSummaries = new Map()
+      for (const workflow of this.workflows) {
+        const states = new Map()
+        for (const taskProxy of workflow.taskProxies) {
+          for (const job of taskProxy.jobs) {
+            if (!states.has(job.state)) {
+              states.set(job.state, new Set())
+            }
+            states.get(job.state).add(`${taskProxy.name}.${taskProxy.cyclePoint}`)
+          }
+        }
+        workflowSummaries.set(workflow.name, states)
+      }
+      return workflowSummaries
+    }
+  },
+  created () {
+    this.viewID = `GScan(*): ${Math.random()}`
+    workflowService.register(
+      this,
+      {
+        activeCallback: this.setActive
+      }
+    )
+    this.subscribe('root')
+  },
+  beforeDestroy () {
+    workflowService.unregister(this)
+  },
+  methods: {
+    subscribe (queryName) {
+      /**
+       * Subscribe this view to a new GraphQL query.
+       * @param {string} queryName - Must be in QUERIES.
+       */
+      if (!(queryName in this.subscriptions)) {
+        this.subscriptions[queryName] =
+          workflowService.subscribe(
+            this,
+            QUERIES[queryName]
+          )
+      }
+    },
+
+    unsubscribe (queryName) {
+      /**
+       * Unsubscribe this view to a new GraphQL query.
+       * @param {string} queryName - Must be in QUERIES.
+       */
+      if (queryName in this.subscriptions) {
+        workflowService.unsubscribe(
+          this.subscriptions[queryName]
+        )
+      }
+    },
+
+    setActive (isActive) {
+      /** Toggle the isLoading state.
+       * @param {bool} isActive - Are this views subs active.
+       */
+      this.isLoading = !isActive
+    },
+
+    getWorkflowIcon (status) {
+      switch (status) {
+        case 'running':
+          return 'mdi-play-circle'
+        case 'held':
+          return 'mdi-pause-octagon'
+        default:
+          return 'mdi-help-circle'
+      }
```

Comment:
@oliver-sanders, @hjoliver also added an extra commit to choose the workflow icon. Given we have the icons loaded anyway, and they fit the CSS styles, would it be fine? Or do we need the component with SVG? They don't look exactly like the design sketch, but looks good to me.

![image](https://user-images.githubusercontent.com/304786/68551444-e8192900-0471-11ea-96ec-2bcc85205e54.png)

(did a `cylc hold five` to test the other icon)

Reply:
(note that the Workflows menu item is activated as it matches the "/workflows*" route)

Reply:
They play icons look fine to me, and we can change to custom svg later on if we want to.